### ISSUE TITLE
feat: Remove comments

### ISSUE THREAD

**Issue by KrunchMuffin** [yamadashy/repopack#2]
Would removing comments based on the language be beneficial?
So for Python remove anything that starts with `# `

**Comment by yamadashy** [comment]
Hi there,
Thanks for the great suggestion!

We think this is a valuable feature that could benefit many users. We're planning to implement it in an upcoming version. Here's what we're considering:

- Add a new `removeComments` option (default: false) in the config. 
- Implement comment removal for various languages (starting with Python, JavaScript, HTML, CSS, and more)

We're still working out the details, but we're thinking along these lines. What do you think? Any additional ideas or specific use cases you had in mind?

**Comment by yamadashy** [comment]
Based on your suggestion, I've implemented the comment removal feature. You can check out the pull request here:

https://github.com/yamadashy/repopack/pull/4

Key changes:
- Added a `removeComments` option to the configuration
- Implemented comment removal for various file types (e.g., Python, JavaScript, HTML, CSS)

I'd appreciate your feedback on this implementation. Please let me know if it aligns with your original idea or if you have any suggestions for further improvements.

**Comment by KrunchMuffin** [comment]
Thanks. that was quick. I just tested it. Almost 100%. it removed these comments

```
    """
    Check the current memory usage of the system.

    Args:
        threshold (int): The memory usage percentage threshold for logging a warning.

    Returns:
        float: The current memory usage as a percentage.
    """
```

but not these

```
        if len(self.caches) >= self.max_manufacturers:
            # Remove least recently used cache
            lru_manufacturer = min(self.last_accessed, key=self.last_accessed.get)
```

**Comment by yamadashy** [comment]
Thank you for testing it so quickly! I'll verify the issues you reported when I get home.

**Comment by yamadashy** [comment]
I've run this code locally, and it appears that the lines with # comments are being removed correctly as well.
```
        if len(self.caches) >= self.max_manufacturers:
            # Remove least recently used cache
            lru_manufacturer = min(self.last_accessed, key=self.last_accessed.get)
```

I'm trying to pinpoint the cause of the issue. Would you be able to share some of the code, even if it's just a portion? That would be really helpful in identifying the problem.

**Comment by KrunchMuffin** [comment]
just tried a different project and it worked, so may have been something I messed up.

**Comment by yamadashy** [comment]
I've tested this feature with a couple of libraries, and it appears that there are indeed some cases where # comments are not being removed as expected.
I've specifically looked at:
https://github.com/ethereum/web3.py
https://github.com/dateutil/dateutil

I'll investigate this issue further to understand the root cause and improve the comment removal functionality.
Thank you for bringing this to my attention!

**Comment by yamadashy** [comment]
I've addressed this issue with a PR that has been merged: #14
While fixing the comment removal, we discovered that it was inadvertently removing empty lines as well. 
This has been corrected, which might result in different output than before.

Additionally, I've included a separate fix for the empty line removal issue: #15

These changes have been released in version 1.0.12.

Thank you for bringing this to our attention! Please try the new version and let me know if you encounter any issues.

**Comment by yamadashy** [comment]
Since the fix has been implemented and merged, I'm going to close this issue.
Thanks for your feedback and help in improving Repopack!

Please feel free to open a new issue if you encounter any further problems or have any new suggestions.
